### ISSUE TITLE
Blink default green wallpaper before setting new one?

### ISSUE THREAD

**Issue by tmpm697** [ifreund/river#323]
I've set wallpaper as below:
```
riverctl spawn swaybg -i <correct existing image jpg path> -m fill
```
When I first start pc or just `killall river` in other `tty`, I can observer flash/blink default green wallpaper before the image that I set via swaybg actually display.

How to avoid this?

**Comment by Leon-Plickat** [comment]
It is currently not possible to avoid this.

**Comment by tmpm697** [comment]
Please support this as the usage of setting wallpaper almost useless because of this.

**Comment by Leon-Plickat** [comment]
This is not rivers problem to solve. To change the wallpaper you probably kill swaybg and start a new instance of it. River will of course display its background colour in the time between.

If seamlessly changing wallpapers is an important use case for you, instead you should use a wallpaper client that can change the image it displays will running.

**Comment by lgflorentino** [comment]
I've also experienced similar. I used the example init and added a custom bg colour. I started river with the new settings and it started with the default background colour and did not update to custom background colour until I started a terminal session. Is there any indication where the issue might be?

**Comment by tmpm697** [comment]
@lgflorentino check #318, seem you're asking for different issue.

**Comment by Leon-Plickat** [comment]
River is always configured at runtime, which means there will be a short period of time where river runs totally unconfigured. That is just a consequence of its design.

Here is how that "flash of green" happens:

1. River is started and during startup outputs are created. For every output, one frame is rendered, which will use rivers default background colour (rivers default colours are solarized dark).
2. River executes your `init` script. In this script you set the new background colour or start `swaybg`.
3. River renders a new frame, either to update the background colour or because the surface of `swaybg` appears. This time the background is how you want it.

It could technically be solved, for example by holding rendering until the init script is done (@ifreund ideas?), but I don't think anything like that is planned right now. A good workaround could also be to set rivers default background colour to something less intrusive, like plain black.

---

@lgflorentino what you describe, the background image only changing when a window opens, sounds like a damage bug.

**Comment by tmpm697** [comment]
@Leon-Plickat : Is there a way to hardcode wallpaper in code? even that I don't want flickers when I set it normally.

**Comment by Leon-Plickat** [comment]
@tmpm697 In `river/Config.zig` on line 37 is the default background colour.  Just set all the numbers to `0.0` and it will be black on startup, so the flash is not visible.

**Comment by ifreund** [comment]
> I've also experienced similar. I used the example init and added a custom bg colour. I started river with the new settings and it started with the default background colour and did not update to custom background colour until I started a terminal session. Is there any indication where the issue might be?

Thanks for the report @lgflorentino, fixed in 5daec34.
